extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -370,7 +373,7 @@ public void stopGracefully()
     );
 
     if (workerTaskLauncher != null) {
-      workerTaskLauncher.waitForWorkerShutdown();
+      workerTaskLauncher.stop(true);
```

Comment:
`waitForWorkerShutdown` might still be needed The closer to wait for cancel to be sent to all tasks, since the closer also cleans up the overlord client after this call, causing the reverse problem of workers running without the controller.

Reply:
On looking further, this should not be an issue since stop is a blocking call